crm_deduplicate_by_fuzzy_name/wizards/partner_merge.py

```diff
@@ -0,0 +1,129 @@
+from openerp.exceptions import Warning as UserError
+
+
+class BasePartnerMergeAutomaticWizard(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    # The field is named this way to not be automatically added as a "group by"
+    # option, which is performed in the code through the name of the field
+    fuzzy_name_group_by = fields.Boolean(
+        string='Similar name',
+        help="WARNING: This option is significantly slower than the others, "
+             "because the match cannot be performed in group.")
+    max_fuzzy_name_difference = fields.Integer(
+        string="Max. characters difference", default=2)
+
+    def _register_hook(self, cr):
+        cr.execute(
+            """
+            SELECT *
+            FROM pg_extension
+            WHERE extname = 'fuzzystrmatch'
+            """)
+        if not cr.fetchall():
+            try:
+                # Try to install the extension
+                cr.execute("CREATE EXTENSION fuzzystrmatch")
+            except:
+                raise UserError(
+                    _("PostgreSQL extension 'fuzzystrmatch' is not present on "
+                      "your DB and can't be auto-installed. Please install "
+                      "it."))
+        return super(BasePartnerMergeAutomaticWizard,
+                     self)._register_hook(cr)
+
+    def _compute_selected_groupby(self, this):
+        if this.fuzzy_name_group_by:
+            try:
+                return super(BasePartnerMergeAutomaticWizard,
+                             self)._compute_selected_groupby(this)
+            except osv.except_osv:
+                # There are no groups
+                return []
+        return super(BasePartnerMergeAutomaticWizard,
+                     self)._compute_selected_groupby(this)
+
+    @api.multi
+    def _process_query(self, query):
+        if self.fuzzy_name_group_by:
+            partner_obj = self.env['res.partner']
+            line_obj = self.env['base.partner.merge.line']
+            groups = self._compute_selected_groupby(self)
+            models = self.compute_models()
+            counter = 0
+            # Reuse WHERE part of the built query
+            index_where = query.find('WHERE')
+            if index_where > 0:
+                index_group_by = query.find('GROUP BY')
+                if index_group_by > 0:
+                    where_query = query[index_where + 5:index_group_by]
+                else:
+                    where_query = query[index_where + 5:]
+                self.env.cr.execute(
+                    "SELECT id FROM res_partner "
+                    "WHERE active = True AND (%s)" % where_query)
+                partners = partner_obj.browse(
+                    [x[0] for x in self.env.cr.fetchall()])
+            else:
+                partners = partner_obj.search([])
+            while partners:
+                partner = partners[0]
+                partners -= partner
```

Comment:
No, because this is a dynamic list (see later `partners -= partner_obj.browse(match_partner_ids)`)